dll/win32/iernonce/registry.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * PROJECT:     ReactOS system libraries
+ * LICENSE:     GPL - See COPYING in the top level directory
+ * FILE:        dll\win32\iernonce\registry.cpp
+ * PURPOSE:     Functions to read RunOnceEx registry.
+ * PROGRAMMERS: Copyright 2013-2016 Robert Naumann
+ */
+
+
+#include <cassert>
+#include "registry.h"
+#include <winreg.h>
+
+
+typedef BOOL(*REGENUMCALLBACK)(HKEY hKey, LPCWSTR lpObjectName, DWORD cchszName, PVOID Param);
+
+enum class REG_ENUM_TYPE
+{
+    SUBKEY,
+    VALUE
+};
+
+
+bool RunOnceExSectionCmp::operator()(const pRunOnceExSection &lhs, const pRunOnceExSection &rhs) const
+{
+    return lstrcmpW(lhs->m_SectionName.GetString(), rhs->m_SectionName.GetString()) < 0;
+}
+
+bool RunOnceExEntryCmp::operator()(const ATL::CStringW &lhs, const ATL::CStringW &rhs) const
+{
+    return lstrcmpW(lhs.GetString(), rhs.GetString()) < 0;
+}
+
+BOOL RegEnumObject(HKEY hKey, REG_ENUM_TYPE EnumType, REGENUMCALLBACK Callback, PVOID Param)
+{
+    DWORD dwObjectNum;
+    DWORD dwMaxObjNameLen;
+    LSTATUS Error;
+    switch (EnumType)
+    {
+        case REG_ENUM_TYPE::SUBKEY:
+            Error = RegQueryInfoKeyW(hKey, NULL, 0, NULL,
+                &dwObjectNum, &dwMaxObjNameLen, NULL, NULL, NULL, NULL, NULL, NULL);
+            break;
+        case REG_ENUM_TYPE::VALUE:
+            Error = RegQueryInfoKeyW(hKey, NULL, 0, NULL,
+                NULL, NULL, NULL, &dwObjectNum, &dwMaxObjNameLen, NULL, NULL, NULL);
+            break;
+        default:
+            assert(0);
```

Comment:
Is it expected that we may fall with an entirely different enumeration value here, other than `SUBKEY` and `VALUE`? If not, what's the purpose of that assert here?

Reply:
yes, I guess there won't any other values at least for now.
I will remove this!

Reply:
Yes, there shouldn't be any enumeration other than `SUBKEY` and `VALUE` at least for now. 🤣
I'm adding this because Visual Studio keeping warning me `Error` might not be initialized (a few lines below), so I add this
I think `assert` is just for clarifying what is logically impossible, so I add it here🤣
do I mistaken what `assert` is for ...?